src/components/ItemList.js

```diff
@@ -116,63 +117,22 @@ const ItemList = () => {
           <ul>
             {filteredResults
               ? filteredResults.map((item) => (
-                  <li
-                    key={item.id}
-                    aria-label={
-                      itemStatus(item) === 'inactive'
-                        ? `${item.data.name} is inactive`
-                        : `Need to buy ${item.data.name} ${itemStatus(item)}`
-                    }
-                    className={itemStatus(item).replace(/\s+/g, '')}
-                  >
-                    {' '}
-                    <input
-                      aria-label="purchase item"
-                      type="checkbox"
-                      onChange={() => handleChecked(item.id, item)}
-                      checked={within24Hours(item)}
-                      disabled={within24Hours(item)}
-                    />{' '}
-                    {item.data.name}
-                    <button
-                      type="button"
-                      aria-label={`delete ${item.data.name}`}
-                      onClick={() => handleDelete(item.id, item.data.name)}
-                    >
-                      Delete
-                    </button>
-                  </li>
+                  <Item
+                    item={item}
+                    handleChecked={handleChecked}
+                    handleDelete={handleDelete}
+                    within24Hours={within24Hours}
+                    itemStatus={itemStatus}
```

Comment:
It looks like these props may be concerns of only an item and not the list itself. Could we move these methods within the Item component itself and reduce the API of the Item component and keep the relationship between ItemList <-> Item less dependent on each other?

There may be a good reason for not doing this now: If there's no shared Firestore reference (Context, Redux, Middleware, etc), it may be expensive to instantiate a reference to Firestore for each item in the list. It might be a cool refactor opportunity to have a shared store.

Reply:
I will definitely look into this! Thanks for the feedback!